src/scripts/run_certbot.sh

```diff
@@ -49,12 +49,39 @@ fi
 # $1: The name of the certificate (e.g. domain-rsa)
 # $2: String with all requested domains (e.g. -d domain.org -d www.domain.org)
 # $3: Type of key algorithm to use (rsa or ecdsa)
+# $4: The authenticator to use to solve the challenge
 get_certificate() {
-    info "Requesting an ${3^^} certificate for '${1}'"
+    local challenge_authenticator="$4"
+    local challenge_params=
+    local challenge_type=
+
+    if [ -n "$challenge_authenticator" ] && [[ "$challenge_authenticator" == dns-* ]]; then
```

Comment:
Could probably be shortened to 

```
if [[ "${4}" == dns-* ]]; then 
    ...
else 
   challenge_authenticator="webroot" ...
fi
```

since we have made sure that there will always be something in `$4` from the caller.

Reply:
I actually shortened the variable names a bit, but kept it in `$authenticator` since it's more talking than `$4`, and does not require to rewrite anything here since we're failing when we don't know the authenticator now!

Reply:
Well, I actually went ahead and replaced it by `$4`